### PR TITLE
Feature/jandex parser

### PR DESCRIPTION
**Pull request by HasseNasse**
GOALS:
- Create a simple servlet Jandex parser which locates all annotated classes, given an index file in the classpath. The index file is found using Classgraph, interpreted and Class-info is extracted. Furthermore we filter out annotations that are of importance (in this case, servlet annotations) and register these servlets to the application container
- Create a simple servlet application with maven-shade-plugin to create uber-jar's with RaptorMP, make sure the servlets are loaded and runnable.

NON-GOALS:
- It's not a goal to fix a fully functional jandex parser, we are only interested in servlet support from uber-jar'd deployments.
- it's not a goal to auto-generate jandex index files, this should be added to a seperate tool e.g. `raptormp-maven-plugin`.

### CODE REVIEW

File: server/src/main/java/mp/raptor/server/component/RegisterableServletComponent.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package mp.raptor.server.component;
+
+import io.undertow.servlet.api.ServletInfo;
+import mp.raptor.common.RegisterableComponent;
+
+/**
+ * Purpose:
+ *
```
Comment (HasseNasse) [inline server/src/main/java/mp/raptor/server/component/RegisterableServletComponent.java:25]: Add javadoc

File: server/src/main/java/mp/raptor/server/component/RegisterableServletComponent.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * @author Hassan Nazar
+ * @author www.hassannazar.net
```
Comment (HasseNasse) [inline server/src/main/java/mp/raptor/server/component/RegisterableServletComponent.java]: Remove auto generated webpage refs

File: server/src/main/java/mp/raptor/server/component/RegisterableServletController.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.lang.System.Logger.Level.DEBUG;
+import static java.lang.System.Logger.Level.INFO;
+
+/**
+ * Purpose:
+ *
```
Comment (HasseNasse) [inline server/src/main/java/mp/raptor/server/component/RegisterableServletController.java:46]: Write javadoc

File: server/src/main/java/mp/raptor/server/parser/AnnotationParser.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * Purpose:
+ * Annotation parser and filtering utilites, used by AnnotationReader.
+ *
+ * @author Hassan Nazar
+ */
+public class AnnotationParser {
+
+  private static final System.Logger LOGGER = System.getLogger(AnnotationParser.class.getSimpleName());
+
+  Collection<ClassInfo> parseClassInformation (final InputStream inputStream) throws IOException {
```
Comment (HasseNasse) [inline server/src/main/java/mp/raptor/server/parser/AnnotationParser.java]: Javadoc

File: server/src/main/java/mp/raptor/server/parser/AnnotationParser.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    // Parse Jandex file
+    LOGGER.log(DEBUG, "Attempting to parse jandex file.");
+    final var indexReader = new IndexReader(inputStream);
+    final var index = indexReader.read();
+    return index.getKnownClasses();
+  }
+
+  List<ClassInfo> filterClassesOfType (final Collection<ClassInfo> annotatedClasses, final AnnotationTypes type) {
```
Comment (HasseNasse) [inline server/src/main/java/mp/raptor/server/parser/AnnotationParser.java]: Javadoc

**Review by HasseNasse — COMMENTED** [review]
Some notes to myself before merging

- Add copyright clause to new classes

File: common/src/main/java/mp/raptor/common/logs/LogFormatter.java [hunk 6]
```diff
@@ -35,7 +35,8 @@
  * @see also CoreConsoleLogger LoggerFinder
  */
 public final class LogFormatter {
-  private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss:SSS");
+  private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter
+      .ofPattern("yyyy-MM-dd HH:mm:ss:SSS");
```
Comment (cnygaard) [inline common/src/main/java/mp/raptor/common/logs/LogFormatter.java:39]: Should we take config value of date pattern?

File: server/src/main/java/mp/raptor/server/parser/WebServletAnnotationProcessor.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+   * @param classLocation dot separated class URI
+   * @return Webservlet annotation with detail info
+   */
+  public static WebServlet getServletAnnotation(final String classLocation) {
+    try {
+      final var clazz = Thread.currentThread().getContextClassLoader().loadClass(classLocation);
+      final var annotations = clazz.getAnnotations();
+
+      for (final Annotation annotation : annotations) {
+        if (annotations[0] instanceof WebServlet) {
```
Comment (cnygaard) [inline server/src/main/java/mp/raptor/server/parser/WebServletAnnotationProcessor.java:49]: Is this always annotation[0]?

**Review by cnygaard — APPROVED** [review]
Good

File: server/src/main/java/mp/raptor/server/parser/WebServletAnnotationProcessor.java (hunk 7 above)
Comment (HasseNasse) [reply server/src/main/java/mp/raptor/server/parser/WebServletAnnotationProcessor.java:49]: I'll fix that

**Pull request merged**